src/server/hset_family.cc

```diff
@@ -772,24 +777,16 @@ void HSetFamily::HDel(CmdArgList args, ConnectionContext* cntx) {
   };
 
   OpResult<uint32_t> result = cntx->transaction->ScheduleSingleHopT(std::move(cb));
-  if (result || result.status() == OpStatus::KEY_NOTFOUND) {
-    cntx->SendLong(*result);
-  } else {
-    cntx->SendError(result.status());
-  }
+  if (result.status() == OpStatus::KEY_NOTFOUND)
+    result = OpResult<uint32_t>(0);
+  HSetReplies{cntx}.Send(result);
 }
 
 void HSetFamily::HLen(CmdArgList args, ConnectionContext* cntx) {
   string_view key = ArgS(args, 0);
-
   auto cb = [&](Transaction* t, EngineShard* shard) { return OpLen(t->GetOpArgs(shard), key); };
 
-  OpResult<uint32_t> result = cntx->transaction->ScheduleSingleHopT(std::move(cb));
-  if (result) {
-    cntx->SendLong(*result);
-  } else {
-    cntx->SendError(result.status());
-  }
+  HSetReplies{cntx}.Send(cntx->transaction->ScheduleSingleHopT(cb));
```

Comment:
also here if key not found should return 0 and not key not found
maybe also add a unit test for this and hdel

Reply:
oh I see that this logic for hlen is in the OpLen implementation, this is confusing, I would add this logic to OpDel as well instead of the change in line 781